Initial guide version (#1)
paxexpress/docs: README.md

# Using of PAX.express

The PAX.express service is used for the delivery of firmware to esp32 paxcounter devices. This is done by using the OTA feature of the [ESP32-paxcounter project](https://github.com/cyberman54/ESP32-Paxcounter). Our fork can be found [here](https://github.com/paxexpress/ESP32-Paxcounter), which is already setup to use PAX.express to deliver firmware updates. The PAX.express cli can be found [here](https://github.com/paxexpress/paxexpress-cli).

## Setup guide
### Tooling 

Please clone the [paxexpress-cli](https://github.com/paxexpress/paxexpress-cli) to your filesystem. 
For installation informations read the [README.md](https://github.com/paxexpress/paxexpress-cli/blob/master/README.md).

### Registration and login

> It's a good idea to switch to the poetry shell now:

Reviewer: If a precompiled binary is used, the reader might never have heard of poetry and is lost at the point

# Using of pax.express

The pax.express service is used for the delivery of firmware to esp32 paxcounter devices. This is done by using the OTA feature of 

> the paxcounter project. Our fork of the repo can be found [here](https://github.com/paxexpress/ESP32-Paxcounter). The pax.express cli can be found [here](https://github.com/paxexpress/paxexpress-cli). Both repositories are used in the discussed setup guide later one. Feel free to clone themen for use.

Reviewer: ```suggestion
the paxcounter project. Our fork of the repo can be found [here](https://github.com/paxexpress/ESP32-Paxcounter). The pax.express cli can be found [here](https://github.com/paxexpress/paxexpress-cli). Both repositories are used in the discussed setup guide later one. Feel free to clone them for use.
```

## Setup guide
### Tooling 

Please clone the [paxexpress-cli](https://github.com/paxexpress/paxexpress-cli)  to your filesystem. We use poetry for python package management so please also intall [poetry](https://python-poetry.org/docs/#installation).

Using poetry change dirctory to the repo's root directory and call:
```

> poetry env use python3.7

Reviewer: I think we can skip this one as the user must have python3.7 installed for this. python 3.6 or 3.8 should be fine as well

> poetry shell

Reviewer: not needed for installation

poetry install
```

after this you should get the cli's command options by using

```

> paxexpress --help

Reviewer: It is unclrear from the docs that this command only works after calling `poetry shell` beforehand.

Maybe use `poetry run` instead?

```
### Registration and login

> Using `authentication` module you can use the register command to register to the pax.express service.

Reviewer: I thought we were to rename it to `auth`?

``` 
paxexpress authentication register
``` 

You will need the provided beta key at this step.

> After registration you can use the login command to get a 30min token for auth against your account which is stored in your keyring. 

Reviewer: ```suggestion
After registration you can use the login command to get a 30min token for auth against your account which is stored in your operating system's password store. 
```

In this file you configure your paxcounter. Please adjust the values accordingly to your wished configuration. Be aware that you will need to receive rcommands at the later part of this guide. E.g. this can be done by using mqtt and a ethernet port.

Now please make a copys of `src/loraconf_sample.h` and rename it to `src/loraconf.h`. Also  make a copys of `src/ota_sample.conf` and rename it to ``src/ota.conf`. In the later file change `BINTRAY_USER` to your user on paxexpress, `BINTRAY_REPO` to your create Repository name and `BINTRAY_API_TOKEN` to your password.

Also `OTA_WIFI_SSID` and `OTA_WIFI_PASS` to a network the OTA will come over. BE AWARE THIS WIFI CREDENTAILS WILL BE UPLOADED TO PAXEXPRESS INSIDE THE FIRMWARE.

Now you are ready to flash the firmware on your board by using platformio tools.

```
pio run -t upload 
```

When you want to upload a version of the firmware for OTA set the
`default_envs` to be `ota` in `platformio.ini`. Now you can use the same commmand to upload it to paxexpress, but you should increment the `release_version` in `platformio.ini` under `common`.

```
pio run -t upload 
```

> Now when sending a rcommand of `[0x9, 0x9]` or `CQk=` over your rcommand channel of choice a OTA should be triggered on the device.

Reviewer: clarify that `CQk=` is for mqtt, not lora

> # Using of pax.express

Reviewer: ```suggestion
# Using of PAX.express
```

The PAX.express service is used for the delivery of firmware to esp32 paxcounter devices. This is done by using the OTA feature of 

> the paxcounter project. Our fork of the repo can be found [here](https://github.com/paxexpress/ESP32-Paxcounter). The PAX.express cli can be found [here](https://github.com/paxexpress/paxexpress-cli). Both repositories are used in the discussed setup guide later one. Feel free to clone them for use.

Reviewer: ```suggestion
the [ESP32-paxcounter project](https://github.com/cyberman54/ESP32-Paxcounter). Our fork can be found [here](https://github.com/paxexpress/ESP32-Paxcounter), which is already setup to use PAX.express to deliver firmware updates.
The PAX.express cli can be found [here](https://github.com/paxexpress/paxexpress-cli).
```

## Setup guide
### Tooling 

> Please clone the [paxexpress-cli](https://github.com/paxexpress/paxexpress-cli) to your filesystem. We use poetry for python package management so please also intall [poetry](https://python-poetry.org/docs/#installation).

Reviewer: just link to the readme for the CLI how to install it

Using poetry change dirctory to the repo's root directory and call:
```
poetry install
```

after this you should get the cli's command options by using

```
peotry run paxexpress --help
```
### Registration and login
It's a good idea to switch to the poetry shell now:

```
peotry shell
```

Using `auth` module you can use the register command to register to the PAX.express service.

``` 
paxexpress auth register
``` 

You will need the provided beta key at this step.

After registration you can use the login command to get a 30min token for auth against your account which is stored in your operating system's password store. 
``` 
paxexpress auth login
```

When you want to remove your auth informations use. 

``` 
paxexpress auth logout
```

### Repositories and packages

> PAX.express uses the structure of repositories which contain packages

Reviewer: please no line breaks within a sentence

Now please make a copy of `src/loraconf_sample.h` and rename it to `src/loraconf.h`. Also  make a copy of `src/ota_sample.conf` and rename it to `src/ota.conf`. In the later file change `BINTRAY_USER` to your user (not your email you using for login!) on PAX.express, `BINTRAY_REPO` to your created Repositorys' name and `BINTRAY_API_TOKEN` to your password.

Also `OTA_WIFI_SSID` and `OTA_WIFI_PASS` needs to be set to a network the OTA will downloaded over. BE AWARE THIS WIFI CREDENTAILS WILL BE UPLOADED TO PAXEXPRESS INSIDE THE FIRMWARE AND ARE AVALIBLE PUBLICLY ON THE SERVICE AT THIS POINT.

Now you are ready to flash the firmware on your board by using platformio tools.

```
pio run -t upload 
```

When you want to upload a version of the firmware for OTA set the
`default_envs` to be `ota` in `platformio.ini`. Now you can use the same commmand to upload it to PAX.express, but you should increment the `release_version` in `platformio.ini` under `common`.

```
pio run -t upload 
```

> Now when sending a rcommand of `[0x9, 0x9]` (or `CQk=` on a mqtt publish) over your rcommand channel of choice a OTA should be triggered on the device.

Reviewer: ```suggestion
Now the rcommand to trigger OTA can be used: `0x09 0x09]` via LoRa or `CQk=` via mqtt
```
